### DISCUSSION TITLE
Flutter build instructions missing

### DISCUSSION THREAD

**Discussion by Root-Core** [rustdesk/rustdesk#4528] [Q&A]
Hello, I wanted to improve some things but struggle to compile the project.

After running `VCPKG_ROOT=$HOME/vcpkg cargo run` I was welcomed by the old interface, created with sciter which confused me a bit.. after some research I understood it was ported to flutter, but there are little instructions how to actually build it (or I just can't find them).

What I did (on Manjaro):
```
yay -Ss flutter
export PATH="$PATH:/opt/flutter/bin"
python3 ./build.py --flutter
```

This still fails:

```
features: ['flutter', 'flutter_texture_render']
False
   Compiling rustdesk v1.2.0 (/home/aede/Workspace/rustdesk)
error: failed to run custom build command for `rustdesk v1.2.0 (/home/aede/Workspace/rustdesk)`

Caused by:
  process didn't exit successfully: `/home/aede/Workspace/rustdesk/target/release/build/rustdesk-8843453e18de07eb/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=Cargo.toml
  cargo:rerun-if-changed=src/flutter_ffi.rs

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: flutter is not a command, or not executable.', build.rs:108:43
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error occurred when executing: cargo build --features flutter,flutter_texture_render --lib --release. Exiting.
```

I would kindly ask if you can update / provide the documentation for building the new UI.

**Comment by deep-soft** [comment]
For flutter build you need some extra steps compared with sciter version.
You can open:
https://github.com/rustdesk/rustdesk/blob/master/.github/workflows/flutter-build.yml
and extract the commands from there.
You can see the build process here:
https://github.com/rustdesk/rustdesk/actions/runs/5131132465

**Reply by Root-Core** [reply]
Thank you, but the point still remains: There should be current instructions as new contributors will stumble upon that problem and the documentation is severely out of date in this regard. Also the whole `build.py` process is not documented yet. So every new contributor has to work out how to get it to work.

I have indeed used the `flutter-build.yml` to extract some commands, but it is a pretty big file and I hadn't had the time. Also it is not targeted for Arch, so it is a bit different.

The best "documentation" might be the `PKGBUILD` for `rustdesk-git`:
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=rustdesk-git

**Reply by huapage** [reply]
> For flutter build you need some extra steps compared with sciter version. You can open: https://github.com/rustdesk/rustdesk/blob/master/.github/workflows/flutter-build.yml and extract the commands from there. You can see the build process here: https://github.com/rustdesk/rustdesk/actions/runs/5131132465

From https://github.com/rustdesk/rustdesk/actions find nightly build logs.

Then follow the log to try to build on tag 1.3.7 ( win32 x64 target, visual studio 2022 )
...
vcpkg install libvpx:x64-windows-static libyuv:x64-windows-static opus:x64-windows-static aom:x64-windows-static ffmpeg[amf,core,nvcodec,qsv]:x64-windows-static
...

python.exe ./build.py --portable --hwcodec --flutter --vram --skip-portable-pack

Get the following errors:

error[E0583]: file not found for module `bridge_generated`
  --> src\lib.rs:33:1

error[E0277]: the trait bound `EventToUI: IntoIntoDart<_>` is not satisfied
  --> src\flutter.rs:1108:32
( total 5 [E0277] errors )

Not sure what is the reason. Possible related to flutter version?

**Comment by rustdesk** [comment]
#763
